docs/devel/luns.md

```diff
@@ -0,0 +1,56 @@
+Direct LUNs migration
+==============
+
+Summary
+-------
+
+VM contains disks. Some of them might be connected as a direct LUN.
+When a user wish to re-use the LUN disks, he needs them attached to the VM.
+The LUN should be used in a single place, or else, it will cause a data corruption.
+The storage where the LUNs reside, needs to be reachable to the VM environment.
+This feature page mostly about import from oVirt.
+
+Design
+------
+
+The requirements for the LUN migration code are (not in priority order)
+
+1. Gather the LUN informnation from the source destination.
+2. To ease on the user - less manual actions, more automated migration.
+3. Save unnecessary risks.
+
+Requirements
+----------------------------------------
+
+1. Destination (OCP cluster) reachability to the ISCSI storage, where the LUN is.
+2. RBAC permission to create `PersistentVolumes` by the controller.
+
+Risks and Mitigations
+----------------------------------------
+
+1. A LUN being used by a VM the user wish to migrate, without reachability to the destination.
+2. A VM that will be used while or after the migration with a direct LUN connected.
+   - A warning should be shown before starting the migration.
```

Comment:
we need to block starting of the original VM in some way. We can't risk corruption

Reply:
We can't really own that VM on source. We can detach the LUN from the VM. WDYT?

Reply:
well - we have to modify it in this case. Not sure if it's worth to provide an option for not detaching it. Maybe for development/testing

Reply:
+1 for detaching it at the end of the migration, right before the vm on the destination cluster can be started

Reply:
Sometimes the VM may start before the migration ends. We need to be delicate here.

Reply:
yeah, the vm might start on oVirt during the migration but as long as it can't start on Openshift Virtualization, we're safe, and it shouldn't be possible to initiate a VMI until the data volumes are ready - so if we can detach the LUN on the source at the end of the migration, before switching the data volumes to ready, it would be fine

Reply:
i think it might be a hard catch.
the importer executing the copyDisk. we check when it's done. I think when it is - the VM might be available to use even when the migration didn't complete. the reconcile will see when it's done and pass to the next phase. overall i think if the user manually "on it" he may have a few seconds to start the vm before the migration of that vm is done in mtv.

Reply:
if there's no "hook" to plug this to CDI before switching the data volumes to 'ready' state, I'd do it once the vm on the source is down during warm migration and at the beginning of cold-migration